File: player/src/tiny/scene/layer.cpp
#include <tiny/scene/layer.h>
#include <tiny/scene/node.h>

Layer::Layer(const uint8_t *_definition, Node **_nodes, uint8_t _numberOfNodes) :
    definition(_definition), nodes(_nodes), numberOfNodes(_numberOfNodes) {
}

void Layer::draw() {
  for(uint8_t i = 0; i < numberOfNodes; i++) {
    Node *node = nodes[i];
    node->draw();
  }
}

File: export/lib/model/project/entities/scene/layer.js
const Entity = require('../entity');
const Nodes = require('./layer/nodes');

class Layer extends Entity {

  get baseType() {
    return 'scene/layer';
  }

  link() {
    super.link();
    this.scene = this.parent;
    this.index = this.scene.layers.models.indexOf(this);
    this.nodes = new Nodes(this, this.models.filter(model => model.baseType === 'scene/layer/node'));
    this.nodes.link();
  }

}

module.exports = Layer;
